### ISSUE TITLE
Rename formulation takes forever

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/PK-Sim#2477]
In the project from https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2466: 
Try to rename the formulation **NoyesWhitney_Polydispers**

After 5 minutes of waiting I have shutdown PK-Sim

**Comment by PavelBal** [comment]
Yes, it is because PK-Sim updates all simulations with this BB after renaming, for whatever reason.

**Comment by msevestre** [comment]
It's because the name of the formulation is used to create the model structure. Still, unless you have a lot of simulation, it shouldn't take that long

**Comment by Yuri05** [comment]
there are 3 simulations in the project. another project of the same kind (which I cannot upload) has only 2 simulations. 
it takes only few seconds to load/create those simulations. the (main) reason must be elsewhere

**Comment by msevestre** [comment]
I agree

**Comment by rwmcintosh** [comment]
Some other change must have fixed this. I cannot reproduce the delay. Whether I open or not the simulations, the rename only takes at most a few seconds.

That time is mostly in the lazy load time when the simulation is not opened before renaming.

**Comment by rwmcintosh** [comment]
@Yuri05 could you check it again and close if not reproducible.

**Comment by Yuri05** [comment]
Yep, seems to work now!
